include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1033,58 @@ namespace GridTools
   ( const MeshType                                                                    &mesh,
     const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate );
 
+  /**
+   * Given a cell @p parent_cell of a mesh a predicate and a flag @p has_predicate computes
+   * the bounding box of all descendants cells of @p parent_cell (or the cell itself, if
+   * it is active) which have the predicate property.
+   *
+   * This is an example of the predicate which is needed to find the bounding box of
+   * locally owned cells:
+   * @code
+   * template <int dim>
+   * bool
+   * pred_locally_owned(const typename Triangulation<dim>::active_cell_iterator &cell)
+   * {
+   *   return cell->is_locally_owned();
+   * }
+   * @endcode
+   *
+   * This function doesn't work with curved meshes.
+   */
+  template < class MeshType >
+  BoundingBox < MeshType::space_dimension >
+  compute_cell_predicate_bounding_box
```

Comment:
do you need this function in public domain (available to the user)? I would say no, so I would put it into your `.cc`

Reply:
I agree it's highly unlikely someone will want to use this function by itself, that includes me of course! The current version of compute_mesh_predicate_bounding_box uses it: I created it to make the code more readable. If there's a better way to encapsulate it, or there's some way to hide it from the public, I'm willing to do it.

Also if there was a function such as "cell->is_children(parent)" which worked between different layers I could think about using the already present compute_bounding_box inside compute_mesh_predicate_bounding_box but I couldn't find one...do you know if I'm missing something?

Reply:
> If there's a better way to encapsulate it, or there's some way to hide it from the public

that's exactly what I already suggested above :smile:  declare and use it from your `grid_tools.cc` file, say
```
namespace internal
{
    //blabla
}
```

> do you know if I'm missing something?

that i don't know